src/metabase/native_query_analyzer.clj

```diff
@@ -49,11 +49,26 @@
    ;; (t2/table-name :model/Table) doesn't work on CI since models/table.clj hasn't been loaded
    :join [[:metabase_table :t] [:= :table_id :t.id]]})
 
+;; NOTE: be careful when adding square braces, as the rules for nesting them are different.
+(def ^:private quotes "\"`")
+
+(defn- quote-stripper
+  "Construct a function which unquotes values which use the given character as their quote."
+  [quote-char]
+  (let [doubled (str quote-char quote-char)
+        single  (str quote-char)]
+    #(-> (subs % 1 (dec (count %)))
+         (str/replace doubled single))))
+
+(def ^:private quote-strippers
+  "Pre-constructed lambdas, to save some memory allocations."
```

Comment:
This may seem a bit OTT, just saving allocating two strings and a function, but it's no extra code to use this map rather than check whether the first character is a quote and then run a stripping function so figured why not.